File: python/0674最长连续增序列.py

# 双指针
def findLengthOfLCIS(nums: list) -> int:
    l=0
    r=0
    res=0
    size=len(nums)

    while(r<size):
        r+=1
        res=max(res,r-l)
        if(r>=size):
            break
        if(nums[r]<=nums[r-1]):
            l=r

    return res

# dp
def findLengthOfLCIS1(nums: list) -> int:
    dp=1
    res=1
    for i in range(1,len(nums)):

        if(nums[i]>nums[i-1]):
            dp+=1
            res=max(res,dp)
        else:
            dp=1
    
    return res

print(findLengthOfLCIS( [] ))

File: csharp/0674最长连续增序列.cs

public partial class Solution
{
    public int FindLengthOfLCIS(int[] nums)
    {
        int size = nums.Length;
        int[] dp = new int[size];

        for (int i = 0; i < size; i++)
        {
            dp[i] = 1;
        }

        for (int i = 1; i < size; i++)
        {
            if(nums[i]>nums[i-1])
                dp[i] += dp[i - 1];
        }

        return dp.Max();
    }
}